packages/fs-detectors/src/workspaces/get-workspace-package-paths.ts

```diff
@@ -87,29 +90,51 @@ async function getPackagePaths(
 async function getPackageJsonWorkspacePackagePaths({
   fs,
 }: GetPackagePathOptions): Promise<string[]> {
-  const packageJsonAsBuffer = await fs.readFile('package.json');
-  const { workspaces } = JSON.parse(
-    packageJsonAsBuffer.toString()
-  ) as PackageJsonWithWorkspace;
-
-  let packages: string[] = [];
-
-  if (Array.isArray(workspaces)) {
-    packages = workspaces;
-  } else {
-    packages = workspaces?.packages ?? [];
+  try {
```

Comment:
In what case would this throw? It seems bad to consume all errors without any logging.

Reply:
If the file doesn't exist it would throw

Reply:
BUT actually I don't think this would ever happen because we are already checking for the file existence when checking which type of monorepo manager it is

Reply:
`packages/fs-detectors/test/fixtures/43-nx-json-misshaped/workspace.json` I added this example when no projects exists